src/oph/ehoks/virkailija/system_handler.clj

```diff
@@ -250,7 +251,12 @@
       (assoc
         (if-let [hoks (hoks/get-with-hankittavat-koulutuksen-osat! hoks-id)]
           (if (= :odottaa-kasittelya
-                 (op/initiate-if-needed! :aloituskysely hoks {:resend? true}))
+                 (op/initiate-if-needed!
+                   {:hoks           hoks
+                    :opiskeluoikeus (koski/get-existing-opiskeluoikeus!
```

Comment:
Näis varmaan pystyis hyödyntämään jotain `wrap-opiskeluoikeus` -koodia

vakavuus: idea

Reply:
Tota middlewarea ennen pitäis olla pinossa `wrap-hoks`, joka nykyisellään hakee pelkän HOKSin päätason. Eli tuo `hoks/get-with-hankittavat-koulutuksen-osat!` pitäis joko purkaa tuohon tai sitten HOKS haetaan kannasta kahteen otteeseen. Jätän tän miettimisen ehkä tän PR:n ulkopuolelle. :thinking: